src/installer.ts

```diff
@@ -27,31 +27,125 @@ if (!tempDirectory) {
   tempDirectory = path.join(baseLocation, 'actions', 'temp');
 }
 
-export class DotnetCoreInstaller {
+/**
+ * Represents the inputted version information
+ */
+export class DotNetVersionInfo {
+  private fullversion: string;
+  private isExactVersionSet: boolean = false;
+
   constructor(version: string) {
-    if (semver.valid(semver.clean(version) || '') == null) {
-      throw 'Implicit version not permitted';
+    // Check for exact match
+    if (semver.valid(semver.clean(version) || '') != null) {
+      this.fullversion = semver.clean(version) as string;
+      this.isExactVersionSet = true;
+
+      return;
+    }
+
+    //Note: No support for previews when using generic
+    let parts: string[] = version.split('.');
+
+    if (parts.length < 2 || parts.length > 3) this.throwInvalidVersionFormat();

```

Comment:
My preference is that we allow single parts (3.x or 2.x), but always recommend the two parts (3.1.x). There are cases where the single part will break, but it gives more flexibility to the consumer. That is something that can be addressed in a later iteration though.

Reply:
3.x is a bit more difficult to implement that 3.1.x because
a) the parsing of the version has to be reworked (not the biggest problem) and
b) https://dotnetcli.blob.core.windows.net/dotnet/release-metadata/releases-index.json just gives you a list of release channels. They must also be compared. The problem is what will you get e.g. for 2.x? The highest version (e.g. now 2.2.8 which is EOL now) or the latest released one (e.g. 2.1.17 which is a LTS; may break builds because of: 2.0.0 released at day 1, 2.2.0 at day 2, than 2.0.1 at day 3 and so on)

I think it would be the best if the single-part version is done later, because of the above concerns and that the priority is to have always the latest version of a release channel (security updates).

Reply:
Thanks, I have added an issue to address this in the future. https://github.com/actions/setup-dotnet/issues/93